src/components/views/admin/BuilderView.tsx

```diff
@@ -1,156 +1,243 @@
+  const [selectedType, setSelectedType] = useState('');
+  const handleTypeChange = (value: string): void => {
+    setSelectedType(value);
+  };
+  const dataArray: any[] = Object.keys(groupedData)
+    .filter((key) => key === selectedType)
+    .map((key) => groupedData[key]);
+  const dimensions = [
+    { value: 'mock_type', label: 'Mock Info' },
+    { value: 'personalInfo', label: 'Personal Info' },
+    { value: 'educationInfo', label: 'Education Info' },
+    { value: 'workInfo', label: 'Work Info' },
+    { value: 'skillsInfo', label: 'Skills Info' },
+    { value: 'projectsInfo', label: 'Projects Info' },
+    { value: 'motivationInfo', label: 'Self-Motivation' },
+    { value: 'referencesInfo', label: 'References Info' },
+    { value: 'cvStatusInfo', label: '' },
+  ];
+
+  // Construct the 'cv' object with filtered and grouped data
+  const candidatesCv: { [key: string]: { [key: string]: any } } = {};
+
+  dimensions.forEach((dimension) => {
+    const type = dimension.value;
+    const dataObj = Object.keys(groupedData)
+      .filter((key) => key === type)
+      .map((key) => groupedData[key]);
+
+    const typeDataByMemberId = DataFilter({
+      dataObject: dataObj,
+      targetType: type,
+    });
+
+    Object.keys(typeDataByMemberId).forEach((memberId) => {
+      if (memberId in candidatesCv) {
+        candidatesCv[memberId][type] = typeDataByMemberId[memberId].map(
+          (item) => item.data,
+        );
+      } else {
+        candidatesCv[memberId] = {
+          [type]: typeDataByMemberId[memberId].map((item) => item.data),
+        };
+      }
+    });
+  });
+
+  const [ratings, setRatings] = useState<{ [key: string]: number | null }>({});
+  const handleRatingChange = (
+    memberId: string,
+    newRating: number | null,
+  ): void => {
+    const rateData = appDataArray.find(
+      (data) =>
+        data.data.memberId === memberId && data.type === 'candidateRateInfo',
+    );
+    if (rateData) {
+      handlePatch(rateData.id, {
+        ...rateData.data,
+        rating: newRating,
+        memberId,
+      });
+    } else {
+      handlePost({ memberId, rating: newRating });
+    }
+    setRatings((prevRatings) => ({
+      ...prevRatings,
+      [memberId]: newRating,
+    }));
+  };
+
+  const handleDownload = async (memberId: string): Promise<void> => {
+    const candidateCvData = candidatesCv[memberId];
+    const { component: CvTemplate } = TEMPLATES.find(
+      (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+    ) || { component: FirstTemplate };
+    const renderedTemplate = <CvTemplate cvValues={candidateCvData} />;
+    const pdfBlob = await pdf(renderedTemplate).toBlob();
+    if (!candidateCvData.cvStatusInfo.customCv) {
+      saveAs(pdfBlob, 'generated-cv.pdf');
+    }
+  };
+  const [visibility, setVisibility] = useState<string | null>(null);
+  const [renderedFile, setRenderedFile] = useState<string | null>(null);
```

Comment:
Where do you use the renderedFile ? if it is not used, then maybe it can just be a variable in your handleView function, since you send the data to a new tab ?

Reply:
Yeh of course, but I used it, to discuss that case, about opening a new tab or not, so we might need it if we didn't want to open a new tab, but if we determined to open a new tab then I will put it as a variable inside handleView function.